pandas/core/dtypes/dtypes.py

```diff
@@ -223,17 +232,24 @@ class CategoricalDtype(PandasExtensionDtype, ExtensionDtype):
     _metadata = ('categories', 'ordered')
     _cache = {}  # type: Dict[str_type, PandasExtensionDtype]
 
-    def __init__(self, categories=None, ordered=None):
+    def __init__(self, categories=None, ordered: bool = None):
```

Comment:
Optional[bool]? (here and elsewhere)

Reply:
The ``= None`` part adds the ``Optional`` part, as per [PEP 484](https://www.python.org/dev/peps/pep-0484/#union-types)

Reply:
From there

> This is no longer the recommended behavior. Type checkers should move towards requiring the optional type to be made explicit.

So it seems like Optional[bool] is preferred.